src/object.rs

```diff
@@ -963,6 +963,20 @@ pub struct WeakRef<T: ?Sized> {
     marker: PhantomData<*const T>,
 }
 
+impl<T> WeakRef<T> {
+    pub fn new() -> Self {
+        unsafe {
+            let weak = WeakRef {
+                inner: Arc::new(WeakRefInner(mem::zeroed())),
+                thread: Some(thread::current().id()),
```

Comment:
None? It should be safe regardless of thread.

Reply:
True